meeteval/io/seglst.py

```diff
@@ -576,47 +584,71 @@ def apply_multi_file(
         h_minus_r = list(set(hypothesis.keys()) - set(reference.keys()))
         r_minus_h = list(set(reference.keys()) - set(hypothesis.keys()))
 
-        ratio = len(r_minus_h) / len(reference.keys())
+        if len(reference.keys()) == 0:
+            logging.warning(
+                f'The reference is empty, but found keys '
+                f'{list(hypothesis.keys())} in the hypothesis'
+            )
+            ratio = 0
```

Comment:
Hmm. What is the motivation to allow an empty reference?
I will run in the `ratio <= allowed_empty_examples_ratio` if statement and only print a warning.

Shouldn't be the check here: `if len(reference.keys()) == 0 and len(hypothesis.keys()) == 0`? But I am not sure, if it is possible to hit this case.

Reply:
Hmm, right, I just wanted to fix this case because it failed in random tests. But it could be better to let this case fail. On the other hand, it could be annoying if it fails for an empty reference. In most cases it'll be a usage error, but there are rare cases where you want an empty reference. 

An alternative would be to disallow empty references and enforce that the user filters the inputs correctly, e.g., with the `--regex` argument

Reply:
I meant disallowing missing keys in the reference, which is currently interpreted as intended by the user and dropped in the hypothesis